### PR TITLE
Adding type system

### PR DESCRIPTION
**Pull request by Vulcanium**
Adding of (true), (false), (and), (app), and (var)

### CODE REVIEW

File: docs/python101/python101-spec.tex [hunk 1]
```diff
@@ -120,7 +120,7 @@ \section{Type system}
 
 $\infer[\text{(and)}]%
 {\Delta;\Gamma \vdash \kw{and}~\textrm{e}_\textrm{1},~\textrm{e}_\textrm{2} \yield \kw{bool}}%
-{\Delta;\Gamma \vdash \textrm{e}_\textrm{1} \colcol \kw{bool}~~ \Delta;\Gamma \vdash\textrm{e}_\textrm{2} \colcol \kw{bool} & \kw{and}:\kw{bool}\rightarrow\kw{bool}\rightarrow\kw{bool}\in \Delta}$
+{\Delta;\Gamma \vdash \textrm{e}_\textrm{1} \colcol \kw{bool}~~ \Delta;\Gamma \vdash\textrm{e}_\textrm{2} \colcol \kw{bool} & \kw{and}:\kw{bool}\times\kw{bool}\rightarrow\kw{bool}\in \Delta}$
```
Comment (fredokun) [inline docs/python101/python101-spec.tex]: Be careful, we are not in a lambda-calculus, the arguments are a cartesian product, Python functions are not currified so it's an \rightarrow.

Comment (fredokun) [reply docs/python101/python101-spec.tex]: so now it's good ;-)

**Comment by fredokun** [comment]
beware of conflicts (they were easy to solve)

**Comment by Vulcanium** [comment]
Merci, j'avais justement hésité entre le \rightarrow et \times. Le \rightarrow est d'ailleurs présent dans la première version de mon PR.


________________________________
De : Frederic Peschanski <notifications@github.com>
Envoyé : vendredi 16 février 2018 18:28
À : nohtyprm/MrPython
Cc : Vulcanium; Author
Objet : Re: [nohtyprm/MrPython] Adding type system (#45)


@fredokun commented on this pull request.

________________________________

In docs/python101/python101-spec.tex<https://github.com/nohtyprm/MrPython/pull/45#discussion_r168820350>:

> @@ -120,7 +120,7 @@ \section{Type system}

 $\infer[\text{(and)}]%
 {\Delta;\Gamma \vdash \kw{and}~\textrm{e}_\textrm{1},~\textrm{e}_\textrm{2} \yield \kw{bool}}%
-{\Delta;\Gamma \vdash \textrm{e}_\textrm{1} \colcol \kw{bool}~~ \Delta;\Gamma \vdash\textrm{e}_\textrm{2} \colcol \kw{bool} & \kw{and}:\kw{bool}\rightarrow\kw{bool}\rightarrow\kw{bool}\in \Delta}$
+{\Delta;\Gamma \vdash \textrm{e}_\textrm{1} \colcol \kw{bool}~~ \Delta;\Gamma \vdash\textrm{e}_\textrm{2} \colcol \kw{bool} & \kw{and}:\kw{bool}\times\kw{bool}\rightarrow\kw{bool}\in \Delta}$


Be careful, we are not in a lambda-calculus, the arguments are a cartesian product, Python functions are not currified so it's an \rightarrow.

**Pull request merged**